src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataPull.java

```diff
@@ -0,0 +1,193 @@
+  private static GitHubDataPull instance() {
+    if (INSTANCE == null) {
+      INSTANCE = new GitHubDataPull();
+    }
+    return INSTANCE;
+  }
+  
+  /**
+   * Creates a single instance of this helper class.
+   * 
+   * @param github An interface to the GitHub API.
+   * @param project of type {@link GitHubProject}, holds GitHub project information.
+   * @return instance of this data pull helper.
+   */
+  public static synchronized GitHubDataPull instance(GitHub github, GitHubProject project) {
+    return instance().github(github).project(project);
+  }
+
+  /**
+   * Sets a variable which should be used by the helper class.
+   *
+   * @param github An interface to the GitHub API.
+   * @return This data pull helper.
+   */
+  public GitHubDataPull github(GitHub github) {
+    this.github = Objects.requireNonNull(
+        github, "Oh no! You gave me a null instead of a GitHub instance!");
+    return this;
+  }
+
+  /**
+   * Sets a variable which should be used by the helper class.
+   *
+   * @param project of type {@linkplain GitHubProject} containing GitHub project information.
+   * @return This data pull helper.
+   */
+  public GitHubDataPull project(GitHubProject project) {
+    this.project =
+        Objects.requireNonNull(project,
+            "Oh no! You gave me a null instead of a GitHubProject instance!");
+    return this;
+  }
+
+  /**
+   * Gets the list of commits for a GitHub project repository.
+   * 
+   * @return List of {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  private List<GHCommit> commits() throws IOException {
+    if (commits == null) {
+      commits = reverseSort(repository().listCommits().toList());
+    }
+    return commits;
+  }
+
+  /**
+   * Gets the GitHub project repository.
+   * 
+   * @return {@link GHRepository} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public GHRepository repository() throws IOException {
+    if (repository == null) {
+      repository = github.getRepository(project.path());
+    }
+    return repository;
+  }
+
+  /**
+   * Gets the first commit of the repository.
+   * 
+   * @return {@link GHCommit} with the project information.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  public GHCommit firstCommit() throws IOException {
+    List<GHCommit> commits = commits();
+    return commits.get(commits.size() - 1);
+  }
+
+  /**
+   * Sort the commits collection in the reverse order by commit date. That is the first element is
+   * the latest commit and the last element is the first commit.
+   * 
+   * @return reverse sorted List of type {@link GHCommit}.
+   * @throws IOException occurred during REST call to GitHub API.
+   */
+  private List<GHCommit> reverseSort(List<GHCommit> commits) throws IOException {
+    return commits.stream()
+        .sorted(compareByCommitDate().reversed())
+        .collect(Collectors.toList());
```

Comment:
I tried both. Unfortunately, the `repository.listCommits().toList()` returns an `unmodifiable` list. Which will not be sorted with these 2 techniques.

Reply:
This should work:

```java
new ArrayList<>(repository.listCommits().asList()).sort(Collections.reverseOrder(BY_COMMIT_DATE));
```

I don't have a proof that it works faster than streams. The code above copies a list and then sort it. It's pretty simple and straightforward. Streams are more complex by may be more elegant. So, feel free to use streams if you like. But please replace `compareByCommitDate().reversed()` with a `static final`:

```java
private static final Comparator<GHCommit> BY_COMMIT_DATE = (firstCommit, secondCommit) -> {
      try {
        return firstCommit.getCommitDate().compareTo(secondCommit.getCommitDate());
      } catch (IOException e) {
        throw new UncheckedIOException("Could not get date from the commit.", e);
      }
    };
```
It's not necessary to create a new comparator each time since it's stateless.